### DISCUSSION TITLE
Html section response is null

### DISCUSSION THREAD

**Discussion by Hame313** [EvgeniyMukhamedjanov/liquid-ajax-cart#27] [Q&A]
Could you help me with this error please?

**Comment by Hame313** [comment]
Also the quantity of items doesnt  work right and the items do not load after added to cart and they need to be refreshed to display them, hope someone can help me im a beginner

**Reply by EvgeniyMukhamedjanov** [reply]
Hey, do you use any of premade themes like Dawn? Or you are making a theme from scratch? 
If you use Dawn for exapmle, it might have a conflict with the Dawn js code.
And can you share preview link to your store so I can see what is going wrong there?

**Reply by Hame313** [reply]
im not using the dawn theme and the preview link is here  https://yrbjxtberlrq5oa0-44114313365.shopifypreview.com

**Reply by EvgeniyMukhamedjanov** [reply]
I've just checked the cart html code and found a strange thing: your cart section has `id=shopify-section-cart-drawerMB` - it has MB at the end. Does this id generated by Shopify or you changed it somehow by yourself?
Liquid Ajax Cart tries to find the `cart-drawerMB` section code but it can't because there is no such section.
![image](https://user-images.githubusercontent.com/13817355/177646193-6cae7726-2454-4445-a3a4-fa0c11750a70.png)

**Comment by Hame313** [comment]
i will change it asap and check if it fixes thank you

**Reply by EvgeniyMukhamedjanov** [reply]
Wait :-) 
You should use the `data-ajax-cart-section` attribute only within a Shopify section, and the id should be generated by Shopify. If you set the id manually -- most likely it will not work.

**Reply by Hame313** [reply]
i've changed it now it only displays Liquid Ajax Cart: the received HTML for the "cart-drawer" section has a different quantity of the "data-ajax-cart-section" containers. The section will be updated completely. this and a problem with money/currency formatter  and i think its because im using the liquid ajax 1.6 version could that be an issue?

**Reply by EvgeniyMukhamedjanov** [reply]
I happened because you specified the id manually and you mentioned another section that doesn't have `data-ajax-cart-section` attribute.
So in order to fix this -- create a new section, lets say `sections/my-floating-cart.liquid`, put all you html code inside the new section (but remove your manual id), and then include this new section to the `theme.liquid` using `{% section 'my-floating-cart' %}`

**Reply by Hame313** [reply]
so i should create a new file called my-floating-cart and put the id on top of the file to work, thank you for the fast reply you're the best

**Comment by EvgeniyMukhamedjanov** [comment]
yeah, create the new file in the `sections` folder. The id is **not** needed - it will be added automatically by shopify. You need to put the `data-ajax-cart-section` attribute only.

**Reply by Hame313** [reply]
i wanted to add a new like progress bar to it  at the top of the file could i be using the request in progress for it to load  the percentage on the progress bar? or should i be using my own script?

**Reply by EvgeniyMukhamedjanov** [reply]
no, just provide the width of the progress bar in the new `my-floating-cart` file. Something like this:
```
<div class="progress-bar" style="width: {{ progress_percentage }}%">
</div>
```
But you need to calculate the progress_percentage variable in the new section file by yourself using `cart.total_price` variable

**Reply by Hame313** [reply]
thank you for the reply. im gonna share this with probably everyone i work for and with, and im gonna use it on every site i can  thank you for everything
